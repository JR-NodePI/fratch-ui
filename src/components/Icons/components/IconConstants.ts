export const IconType = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
} as const;
