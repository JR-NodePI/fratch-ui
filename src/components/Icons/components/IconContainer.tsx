import { c } from '../../../helpers/classNameHelpers';
import { IconType } from './IconConstants';
import { IconProps } from './IconProps';

import styles from './IconContainer.module.css';

const IconContainer = ({
  svg,
  className,
  type = IconType.PRIMARY,
}: IconProps): JSX.Element => {
  return <div className={c(styles.icon, styles[type], className)}>{svg}</div>;
};

export default IconContainer;
