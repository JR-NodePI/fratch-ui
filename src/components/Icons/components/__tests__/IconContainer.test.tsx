import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { IconType } from '../IconConstants';
import IconContainer from '../IconContainer';
import { type IconContainerProps } from '../IconContainerProps';

describe('Button', () => {
  const setup = (
    props: Omit<IconContainerProps, 'children'> = {}
  ): ReturnType<typeof render> =>
    render(
      <IconContainer {...props}>
        {({ iconClassName }): JSX.Element => <svg className={iconClassName} />}
      </IconContainer>
    );

  it('should render properly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  describe('should render with TYPE', () => {
    Object.values(IconType).forEach(type => {
      it(type, () => {
        const { container } = setup({ type });
        expect(container).toMatchSnapshot();
      });
    });
  });

  it('should render with className', () => {
    const { container } = setup({
      className: 'mock_test_class_name',
    });
    expect(container).toMatchSnapshot();
  });
});
