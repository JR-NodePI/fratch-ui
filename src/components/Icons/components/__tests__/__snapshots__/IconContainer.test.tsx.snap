// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button.tsx > should render properly 1`] = `
<div>
  <div
    class="_icon_2be8fc _primary_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > error 1`] = `
<div>
  <div
    class="_icon_2be8fc _error_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > info 1`] = `
<div>
  <div
    class="_icon_2be8fc _info_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > primary 1`] = `
<div>
  <div
    class="_icon_2be8fc _primary_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > secondary 1`] = `
<div>
  <div
    class="_icon_2be8fc _secondary_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > success 1`] = `
<div>
  <div
    class="_icon_2be8fc _success_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > tertiary 1`] = `
<div>
  <div
    class="_icon_2be8fc _tertiary_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > warning 1`] = `
<div>
  <div
    class="_icon_2be8fc _warning_2be8fc"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with className 1`] = `
<div>
  <div
    class="_icon_2be8fc _primary_2be8fc mock_test_class_name"
  >
    <svg />
  </div>
</div>
`;
